creator-node/src/routes/files.js

```diff
@@ -368,6 +373,59 @@ module.exports = function (app) {
    * TODO: It seems like handleResponse does work with piped responses, as seen from the track/stream endpoint.
    */
   app.get('/ipfs/:dirCID/:filename', getDirCID)
+
+  /**
+   * Serve file from FS given a storage path
+   * This is a cnode-cnode only route, not to be consumed by clients. It has auth restrictions to only
+   * allow calls from cnodes with delegateWallets registered on chain
+   * @dev No handleResponse around this route because it doesn't play well with our route handling abstractions,
+   * same as the /ipfs route
+   * @param req.query.filePath the fs path for the file. should be full path including leading /file_storage
+   * @param req.query.delegateWallet the wallet address that signed this request
+   * @param req.query.timestamp the timestamp when the request was made
+   * @param req.query.signature the hashed signature of the object {filePath, delegateWallet, timestamp}
+   */
+  app.get('/file_lookup', async (req, res) => {
+    const { filePath, timestamp, signature } = req.query
+    let { delegateWallet } = req.query
+    delegateWallet = delegateWallet.toLowerCase()
+
+    // no filePath passed in
+    if (!filePath) return sendResponse(req, res, errorResponseBadRequest(`Invalid request, no path provided`))
+
+    // check that signature is correct and delegateWallet is registered on chain
+    const recoveredWallet = recoverWallet({ filePath, delegateWallet, timestamp }, signature).toLowerCase()
+    const creatorNodes = await getAllRegisteredCNodes()
+    const foundDelegateWallet = creatorNodes.some(node => node.delegateOwnerWallet.toLowerCase() === recoveredWallet)
+    if ((recoveredWallet !== delegateWallet) || !foundDelegateWallet) {
+      return sendResponse(req, res, errorResponseUnauthorized(`Invalid wallet signature`))
+    }
+    const filePathNormalized = path.normalize(filePath)
+
+    // check that the regex works and verify it's not blacklisted
+    const match = FILE_SYSTEM_REGEX.exec(filePathNormalized)
+    if (!match) return sendResponse(req, res, errorResponseBadRequest(`Invalid filePathNormalized provided`))
+
+    const { outer, inner } = match.groups
+    if (await req.app.get('blacklistManager').CIDIsInBlacklist(outer)) {
```

Comment:
NIT - cache blacklist manager? not worth it

Reply:
could you explain what you mean by cache it?

Reply:
why request it each time?

Reply:
if there are inner values

Reply:
these values are cached in redis and we look them up now before we serve any gateway requests. i was following that pattern here